vm_network_migration/tests/test_main.py

```diff
@@ -0,0 +1,363 @@
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test main() function
+"""
+
+import json
+import os
+from unittest import mock
+from unittest.mock import patch
+
+import httplib2
+import unittest2 as unittest
+from googleapiclient.errors import HttpError
+from vm_network_migration.vm_network_migration import *
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+
+
+def datafile(filename):
+    """Generate path of the file
+    Args:
+        filename: file name
+
+    Returns: the file path
+
+    """
+    return os.path.join(DATA_DIR, filename)
+
+
+def read_json_file(filename):
+    """Read *.json file
+    Args:
+        filename: json file name
+
+    Returns: a Python object
+
+    """
+    with open(datafile(filename)) as f:
+        res = json.load(f)
+        f.close()
+    return res
+
+
+@patch(
+    "vm_network_migration.vm_network_migration.roll_back_original_instance")  # index: 10
+@patch("vm_network_migration.vm_network_migration.attach_disk")  # index: 9
+@patch("vm_network_migration.vm_network_migration.delete_instance")  # index: 8
+@patch("vm_network_migration.vm_network_migration.create_instance")  # index: 7
+@patch("vm_network_migration.vm_network_migration.get_network")  # index: 6
+@patch("vm_network_migration.vm_network_migration.get_zone")  # index: 5
+@patch("vm_network_migration.vm_network_migration.detach_disk")  # index: 4
+@patch(
+    "vm_network_migration.vm_network_migration.retrieve_instance_template")  # index: 3
+@patch(
+    "vm_network_migration.vm_network_migration.wait_for_operation")  # index: 2
+@patch("vm_network_migration.vm_network_migration.stop_instance")  # index: 1
+@patch("google.auth.default")  # index 0
+class MainFlowLogic(unittest.TestCase):
+    MOCK_CREDENTIALS = mock.Mock(spec=google.auth.credentials.Credentials)
+    errorResponse = httplib2.Response({
+        "status": 404,
+        "reason": "HttpMock response: invalid network"})
+    project = "mock_project"
+    zone = "mock_us_central1_a"
+
+    def test_basic(self, *mocks):
+        mocks[0].return_value = (self.MOCK_CREDENTIALS, self.project)
+        mocks[3].return_value = read_json_file("sample_instance_template.json")
+        mocks[6].return_value = read_json_file("sample_auto_mode_network.json")
+
+        original_instance = "mock_original_instance"
+        new_instance = "mock_new_instance"
+        target_network = "mock_target_network"
+        target_subnetwork = "mock_target_subnetwork"
+        self.assertIsNone(
+            main(self.project, self.zone, new_instance, original_instance,
+                 target_network, target_subnetwork))
```

Comment:
Add one more verification that the new instance is the same as the previous instance.